### PR TITLE
[4.0] [WIP] Add support for TinyMCE content_css

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

TinyMCE has the ability to set the editor theme, however there are 2 parts to this:

1. The editor wrapper (skin)
2. The content itself

Until now, Joomla has only supported setting the Skin and has included its own `editor.css` file.

What this PR does is:

- Remove the old **Template CSS Classes** parameter.
- Adds 2 new parameters (site/admin) to set the Content CSS.
- Maintains support for using a `editor.css` file, only if specified.

### Testing Instructions

1. Open TinyMCE in your Plugin Manager
2. Click on the `Set 0` tab
3. Set the `Site/Administrator Content CSS` option to `dark`

### Actual result

The whole editor is now a dark theme:

screeny

### CODE REVIEW

**Comment by dgrammatiko** [comment]
This is wrong, the tinyMCE theming should follow the regular Joomla overriding way, eg to override a theme you should just place the new theme in the template `css/vendor/...blabla`. No options needed also it's the expected way. Let's do this properly not introducing exceptions. It's my fault I should have done this the right way many years ago when I introduced the theming option.BTW the code already exists here: https://github.com/joomla/joomla-cms/pull/14510

EDIT: So after reading a bit on the new features of tinyMCE and some talk with @C-Lodder I was wrong this is fine! Anyway I was talking about the skin this is about the content, not the same thing apparently

**Comment by brianteeman** [comment]
Doesnt this prevent a template from shipping an editor.css file ?

**Comment by C-Lodder** [comment]
@brianteeman Technically no, but you would have to explicitly set the `Custom CSS Classes` option to `editor.css` if you wanted to use it. Could re-add the auto detection, but will close in favour of: https://github.com/joomla/joomla-cms/pull/29259

**Comment by brianteeman** [comment]
as its closed its not an issue any more as that pr doesnt change template css

**Comment by dgrammatiko** [comment]
@C-Lodder please reopen this one, it covers other aspects and features that are not accessible right now. Just fix it so it will pick up editor.css automatically and we're good

**Comment by C-Lodder** [comment]
@dgrammatiko I'll wait until this is merged

**Comment by Quy** [comment]
@C-Lodder Please reopen. Thanks.

**Comment by opn365** [comment]
I have tested this item :white_check_mark: successfully on 2c8c33257e8511580ede30b9208ab2190041f38f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29257">issues.joomla.org/tracker/joomla-cms/29257</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 2c8c33257e8511580ede30b9208ab2190041f38f

Default + pr:

image

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29257">issues.joomla.org/tracker/joomla-cms/29257</a>.</sub>

**Comment by richard67** [comment]
Unfortunately the pull request here shows a conflict for the `administrator/language/en-GB/plg_editors_tinymce.ini`, so it shouldn't be set RTC (ready to commit) even if it has 2 good tests.

I'll try to solve the conflict later if @C-Lodder wasn't faster.

**Comment by richard67** [comment]
I've solved the conflict. It was just in the language strings so previous tests are still valid.

**Comment by richard67** [comment]
Will set it RTC as soon as I can see the automated tests are passing.

**Comment by C-Lodder** [comment]
Richard, this currently breaks B/C a little

**Comment by richard67** [comment]
> Richard, this currently breaks B/C a little

@C-Lodder What? The PR? Or how I solved the conflict? And what shall we do now?

**Comment by softforge** [comment]
I have tested this item :white_check_mark: successfully on f406d88657fbfad9c28a62fd838a7bbe77eeb6ea<br /><br />I was testing as well but other tester had not filled in in the sheet so its for extra tests!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29257">issues.joomla.org/tracker/joomla-cms/29257</a>.</sub>

**Comment by C-Lodder** [comment]
@richard67 see this https://github.com/joomla/joomla-cms/pull/29257#issuecomment-636092789

Let me revisit it this weekend

**Comment by richard67** [comment]
> @richard67 see this [#29257 (comment)](https://github.com/joomla/joomla-cms/pull/29257#issuecomment-636092789)
> 
> Let me revisit it this weekend

@C-Lodder Ok, thanks in advance. I'll set the "Updates Requested" label so nobody sets RTC. Let me know when I shall remove that label.

File: plugins/editors/tinymce/tinymce.php [hunk 1]
```diff
@@ -305,7 +304,8 @@ public function onDisplay(
 			// If it is not a URL, assume it is a file name in the current template folder
 			else
 			{
-				$content_css = Uri::root(true) . '/templates/' . $template . '/css/' . $content_css_custom;
+				$content_css    = Uri::root(true) . '/templates/' . $template . '/css/' . $content_css_custom;
```
Comment (dgrammatiko) [inline plugins/editors/tinymce/tinymce.php:307]: @C-Lodder this needs update to support the new template mode (child templates) where the assets are moved to `media/templates/site|administrator/templateName/css`

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

**Comment by C-Lodder** [comment]
I'm going to close this as it's been too long since I last worked on it, and I'm too burned out to continue the thought process.

Will leave the branch open it anyone wants to continue with it.

**Pull request closed without merging**